packages/react-i18n/README.md

```diff
@@ -0,0 +1,226 @@
+
+If you provide any of the above options, you must also provide an `id` key, which gives the library a way to store the translation dictionary.
+
+Here’s the example above with component-specific translations:
+
+```ts
+import * as React from 'react';
+import {EmptyState} from '@shopify/polaris';
+import {withI18n, WithI18nProps} from '@shopify/react-i18n';
+
+import en from './locales/en.json';
+import fr from './locales/fr.json';
+
+export interface Props {}
+type ComposedProps = Props & WithI18nProps;
+
+function NotFound({i18n}: ComposedProps) {
+  return (
+    <EmptyState
+      heading={i18n.translate('NotFound.heading')}
+      action={{content: i18n.translate('NotFound.action'), url: '/'}}
+    >
+      <p>{i18n.translate('NotFound.content')}</p>
+    </EmptyState>
+  );
+}
+
+export default withI18n({
+  id: 'NotFound',
+  fallback: en,
+  translations(locale) {
+    if (locale === 'en') {
+      return en;
+    } else if (locale === 'fr') {
+      return fr;
+    }
+  },
+})(NotFound);
+```
+
+```json
+// NotFound/components/en.json
+{
+  "NotFound": {
+    "heading": "Page not found",
+    "action": "Back",
+    "content":
+      "The page you were looking for could not be found. Please check the web address for errors and try again."
+  }
+}
+```
+
+As shown above, we recommend scoping the translation file to the name of the component to prevent potential naming conflicts resulting from typos in the keys you use.
+
+A few other details are worth noting about translation loading and lookup:
+
+- Your `translations` function can be called several times for a given locale. If, for example, the locale is `en-CA`, your function will be called with `en-CA` and `en`, which allows you to load country-specific variations for translations.
+- The `i18n` object supplied to a given component can reference translations at any level of depth using a keypath (for example, `NotFound.heading` in the code above). It can also reference translations in parent components; use this to include common translations around a component that contains most of your application.
+- When `translate` is called, it looks up translations in the following order: explicit translations provided by the component’s `translations` function, then translations from the `fallback` for the component, then the same process in every parent component, from bottom to top, that are also connected with `i18n`.
+- In the case of asynchronous translations, your component will only be able to look up translations from its (and ancestors’) `fallback` translation dictionaries until the translations have loaded.
+
+##### Replacements
+
+Replacements can be provided as key-value pairs following the translation key. Your translations should reference the relevant key names, surrounded by a single set of curly braces:
+
+```ts
+// Assuming a dictionary like:
```

Comment:
it's weird that this code sample is entirely comments, why not just comment the first line?